src/app/(main)/dashboard/page.tsx

```diff
@@ -1,12 +1,25 @@
-import React from "react";
+import { redirect } from "next/navigation";
 import PreVoyageDashboard from "./components/pre-voyage-dashboard/PreVoyageDashboard";
-import VoyageDashboard from "./components/voyage-dashboard/VoyageDashboard";
+import VoyageDashboardPage from "./components/voyage-dashboard/VoyageDashboardPage";
+import { getUser } from "@/utils/getUser";
 
-function DashboardPage() {
+async function DashboardPage() {
   // TODO: Mocked temporary value
   const isVoyageStarted = true;
 
-  return isVoyageStarted ? <VoyageDashboard /> : <PreVoyageDashboard />;
+  const [user] = await getUser();
+
+  const teamMember = user?.voyageTeamMembers.find(
+    (voyage) => voyage.voyageTeam.voyage.status.name === "Active",
+  );
+
+  const teamId = teamMember?.voyageTeamId;
+
+  if (teamId && isVoyageStarted) {
+    redirect(`/dashboard/${teamId}`);
+  }
+
+  return isVoyageStarted ? <VoyageDashboardPage /> : <PreVoyageDashboard />;
```

Comment:
This page should be responsible for the default voyage dashboard. Should only check for existence of an active voyage team here and if they do, redirect to the dashboard page with team id params. In that page you can also check for if voyage is started or not (which you already have). For the default dashboard, just put some placeholder there for now. Also replace the route with the routepaths util.

Something like this

```
const [user] = await getUser();

  const teamMember = user?.voyageTeamMembers.find(
    (voyage) => voyage.voyageTeam.voyage.status.name === "Active"
  );

  if (teamMember) {
    redirect(routePaths.VoyageMemberDashboardPage(teamMember.voyageTeamId.toString()));
  }

  return <div>Default Dashboard</div>;
```

Reply:
oh and you will need to update the routepath in the routepath util file

Reply:
You're going to need to remove the isVoyagestarted check or else it'd never redirect the user to the pre-voyage dashboard.